src/test/java/io/confluent/connect/elasticsearch/ElasticsearchSinkTaskTest.java

```diff
@@ -273,6 +275,81 @@ public void testVersion() {
     assertTrue(task.version().matches("^(\\d+\\.){2}?(\\*|\\d+)(-.*)?$"));
   }
 
+  @Test
+  public void testConvertTopicToDataStream_allowDashes() {
+    final String type = "logs";
+    final String dataset = "a_valid_dataset";
+    props.put(DATA_STREAM_TYPE_CONFIG, type);
+    props.put(DATA_STREAM_DATASET_CONFIG, dataset);
+    setUpTask();
+
+    String startsWithDash = "-dash";
+    SinkRecord record = record(startsWithDash, true, false, 0);
+    task.put(Collections.singletonList(record));
```

Comment:
```suggestion
    task.put(Collections.singletonList(record(startsWithDash, true, false, 0)));
```

can do that everywhere